js/app/libs.js

```diff
@@ -119,11 +173,34 @@ Fliplet.Registry.set('notification-inbox:1.0:app:core', function(data) {
     return instance.poll(options);
   }
 
-  function getNewNotifications(ts) {
-    return Promise.all([
-      instance.unread.count({ createdAt: { $gt: ts } }),
-      instance.unread.count()
-    ]);
+  function getLatestNotificationCounts(ts, options) {
+    function fetchCounts() {
+      return Promise.all([
+        // @TODO Update to use instance.new.count()
+        instance.unread.count({ createdAt: { $gt: ts } }),
+        instance.unread.count()
+      ]);
+    }
+
+    options = options || {};
+
+    return Fliplet.Navigator.Notifications.getBadgeNumber().then(function(badgeNumber) {
+      var countType = notificationsBadgeType === 'unread' ? 'unreadCount' : 'newCount';
+
+      // App badge number has changed. Get the latest counts immediately.
+      if ((typeof badgeNumber === 'number' && badgeNumber !== storage[countType]) || options.forcePolling) {
+        return fetchCounts;
+      }
+
+      // Get notification counts (throttled at 60 seconds)
+      return Fliplet.Cache.get(
+        {
+          expire: 60,
+          key: 'appNotificationsCounts'
```

Comment:
@tonytlwu can you also add `allowExpired: false` to the options? That ensures that once the 60 seconds have expired this runs right away.

Reply:
@squallstar What's the difference that `allowExpired: false` makes? I thought that because it expires in 60 seconds, if I run this after 60 seconds, it would run the fetch function.

Also, are you suggesting we add this to both this function here as well as the function for updating the `appNotificationSeenAt` timestamp?

Reply:
@tonytlwu `allowExpired` allows an expired cache value to be retrieved only up to once while the refresh happens in background. I've actually just realised we don't need the option because we don't need to promise value, we just use the logic to run the fetch up to once every minute.